### PR TITLE
dev-lang/luajit EAPI bump, slots

### PR DESCRIPTION
**Pull request by ILMostro**
Bug: https://bugs.gentoo.org/657722

Adds new `app-eselect/eselect-luajit` package for switching luajit versions/slots

### CODE REVIEW

**Comment by ILMostro** [comment]
files/luajit-205.patch: needed for luajit-205{,-r1}.ebuild to correct Makefile changes upstream
    files/luajit-nosymlinks.patch: needed for <=luajit-204 to correct Makefile discrepencies
    files/luajit-luaver.patch: needed for correctly finding and specifying the version of lua being used
    
    It should be noted that the HOTFIX upstream patches were removed as sources here

**Comment by ILMostro** [comment]
A smaller subset of changes than #9268 , with clean commit history that follows GLEP66.

File: app-eselect/eselect-luajit/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>ilmostro7@gmail.com</email>
+		<name>ILMostro</name>
+	</maintainer>
```
Comment (hydrapolic) [inline app-eselect/eselect-luajit/metadata.xml]: You probably need to add proxy-maint.

Comment (ILMostro) [reply app-eselect/eselect-luajit/metadata.xml]: ahh, yeah; forgot about that.  Thank you

**Comment by ILMostro** [comment]
@hydrapolic Thanks again; the fix was applied almost immediately.  I'm not sure what the plan is going forward with regards to this item.  There were also suggestions about avoiding the use of slotted `dev-lang/luajit`, and only using slotted `dev-lang/lua`.
In the meantime, I think this is good to go.

**Comment by mgorny** [comment]
To be honest, I don't think this is a good idea. Slotted Lua proved pretty much a total failure and is masked for years now (over 4 years, to be more precise). I don't see why we would try to repeat the same failed experiment on LuaJIT before someone figures out how to deal with Lua itself in the first place.

**Comment by ILMostro** [comment]
Slotted lua works just fine for me, and a few others, AFAIK.  There have been other attempts to create slotted lua, but the formatting of the commit messages and the large number of packages of such an undertaking have deemed it unacceptable.

e.g. https://github.com/gentoo/gentoo/pull/9268

**Comment by weedy** [comment]
@mgorny Lua is up to 5.4, I think having the only unmasked version of Lua be so damn old is a larger problem. The current version of Lua posix in the tree doesn't even work with the Lua we have unmasked.

- Lua 5.1 was released on 21 Feb 2006.
- The last release was Lua 5.1.5, released on 17 Feb 2012.

You guys started talking about the fact the current situation was a problem in 2012. https://bugs.gentoo.org/407091 And the gentoo ML has discussions from 2017. 

We really should unmask Lua before the heat death of the universe...

**Pull request closed without merging**